plugins/database/mongodb/mongodb.go

```diff
@@ -140,6 +140,20 @@ func (m *MongoDB) CreateUser(ctx context.Context, statements dbplugin.Statements
 	case err == nil:
 	case err == io.EOF, strings.Contains(err.Error(), "EOF"):
 		// Call getConnection to reset and retry query if we get an EOF error on first attempt.
+		session, err := m.getConnection(ctx)
+		if err != nil {
+			return "", "", err
+		}
+		err = session.DB(mongoCS.DB).Run(createUserCmd, nil)
+		if err != nil {
+			return "", "", err
+		}
+	case strings.Contains(err.Error(), "not master"):
+		// Close connection and reconnect if connected node is not primary.
+		if err := m.Close(); err != nil {
```

Comment:
I don't think we need to call Close `here`, getConnection should take care of closing the session. Do we also want to retry the query afterwards? If so, we might just combine the EOF case and this one into one (similar to what you're doing in `RevokeUser`.

Reply:
Does getConnection really close the session in this case though? What happens if the session is not nil already and the ping succeeds?

Reply:
As far as I can see getConnection calls Connection() in connection_producer.go which only closes the session if the session is not nil and the ping fails. 

The ping won't fail though, as the node is still up (it's just not the primary). I don't see a way to check whether it's master with the ping command.